lxd/daemon.go

```diff
@@ -545,90 +546,91 @@ func (d *Daemon) Init() error {
 	/* Detect user namespaces */
 	runningInUserns = shared.RunningInUserNS()
 
-	/* Detect AppArmor support */
-	if aaAvailable && os.Getenv("LXD_SECURITY_APPARMOR") == "false" {
-		aaAvailable = false
-		aaAdmin = false
+	/* Detect AppArmor availability */
+	_, err = exec.LookPath("apparmor_parser")
+	if os.Getenv("LXD_SECURITY_APPARMOR") == "false" {
 		logger.Warnf("AppArmor support has been manually disabled")
-	}
-
-	if aaAvailable && !shared.IsDir("/sys/kernel/security/apparmor") {
-		aaAvailable = false
-		aaAdmin = false
+	} else if !shared.IsDir("/sys/kernel/security/apparmor") {
 		logger.Warnf("AppArmor support has been disabled because of lack of kernel support")
-	}
-
-	_, err = exec.LookPath("apparmor_parser")
-	if aaAvailable && err != nil {
-		aaAvailable = false
-		aaAdmin = false
+	} else if err != nil {
 		logger.Warnf("AppArmor support has been disabled because 'apparmor_parser' couldn't be found")
+	} else {
+		aaAvailable = true
 	}
 
-	/* Detect AppArmor admin support */
-	if aaAdmin && !haveMacAdmin() {
-		aaAdmin = false
-		logger.Warnf("Per-container AppArmor profiles are disabled because the mac_admin capability is missing.")
-	}
+	/* Detect AppArmor stacking support */
+	aaCanStack := func() bool {
+		contentBytes, err := ioutil.ReadFile("/sys/kernel/security/apparmor/features/domain/stack")
+		if err != nil {
+			return false
+		}
 
-	if aaAdmin && runningInUserns {
-		aaAdmin = false
-		logger.Warnf("Per-container AppArmor profiles are disabled because LXD is running in an unprivileged container.")
-	}
+		if string(contentBytes) != "yes\n" {
```

Comment:
Hm, so should we maybe trim whitespace on the output of `ReadFile()` to exclude any surprises like `  yes\n` or `yes\n\n`?